src/workspaces/LockWorkspaceModal/LockWorkspaceModal.test.tsx

```diff
@@ -0,0 +1,188 @@
+import { renderWithAppContexts as render } from 'src/testing/test-utils';
+import { defaultGoogleWorkspace } from 'src/testing/workspace-fixtures';
+import LockWorkspaceModal from 'src/workspaces/LockWorkspaceModal/LockWorkspaceModal';
+import { WorkspaceWrapper as Workspace } from 'src/workspaces/utils';
+
+type AjaxContract = ReturnType<typeof Ajax>;
+type AjaxExports = typeof import('src/libs/ajax');
+jest.mock('src/libs/ajax', (): AjaxExports => {
+  return {
+    ...jest.requireActual('src/libs/ajax'),
+    Ajax: jest.fn(),
+  };
+});
+
+type ErrorExports = typeof import('src/libs/error');
+jest.mock('src/libs/error', (): ErrorExports => {
+  return {
+    ...jest.requireActual('src/libs/error'),
+    reportError: jest.fn(),
+  };
+});
+
+describe('LockWorkspaceModal', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+
+  it('contains lock button and message if the workspace is unlocked', () => {
+    // Arrange
+    const workspace: Workspace = {
+      ...defaultGoogleWorkspace,
+      workspace: { ...defaultGoogleWorkspace.workspace, isLocked: false },
+    };
+    const props = { workspace, onDismiss: () => {}, onSuccess: () => {} };
+
+    // Act
+    render(<LockWorkspaceModal {...props} />);
+
+    // Assert
+    const lockButton = screen.getByRole('button', { name: 'Lock Workspace' });
+    expect(lockButton).toBeInTheDocument();
+    expect(screen.getByText(/Are you sure you want to lock this workspace/)).toBeInTheDocument;
+  });
+
+  it('contains unlock button and message if the workspace is locked', () => {
+    // Arrange
+    const workspace: Workspace = {
+      ...defaultGoogleWorkspace,
+      workspace: { ...defaultGoogleWorkspace.workspace, isLocked: true },
+    };
+    const props = { workspace, onDismiss: () => {}, onSuccess: () => {} };
+
+    // Act
+    render(<LockWorkspaceModal {...props} />);
+
+    // Assert
+    const unlockButton = screen.getByRole('button', { name: 'Unlock Workspace' });
+    expect(unlockButton).toBeInTheDocument();
+    expect(screen.getByText(/Are you sure you want to unlock this workspace/)).toBeInTheDocument;
+  });
+
+  it('calls onDismiss, onSuccess, and locks unlocked workspace if the lock is toggled', async () => {
+    // Arrange
+    const onDismiss = jest.fn();
+    const onSuccess = jest.fn();
+    const mockLock = jest.fn();
+    const workspace: Workspace = {
+      ...defaultGoogleWorkspace,
+      workspace: { ...defaultGoogleWorkspace.workspace, isLocked: false },
+    };
+    const props = { workspace, onDismiss, onSuccess };
+    asMockedFn(Ajax).mockReturnValue({
+      Workspaces: {
+        workspace: jest.fn().mockReturnValue({
+          lock: mockLock,
+        }),
+      },
+    } as DeepPartial<AjaxContract> as AjaxContract);
+
+    // Act
+    render(<LockWorkspaceModal {...props} />);
+    const lockButton = screen.getByRole('button', { name: 'Lock Workspace' });
+    await act(async () => {
+      fireEvent.click(lockButton);
```

Comment:
This works, but we prefer to use `await user.click(lockButton)`, where `const user = userEvent.setup();`